### PR TITLE
Add support for cookie-parser

### PR DESCRIPTION
**Pull request by jdtzmn**
Added support for [cookie-parser](https://expressjs.com/en/resources/middleware/cookie-parser.html) middleware

PS: 🥂to how cleanly your code is written.

### CODE REVIEW

**Comment by jaw187** [comment]
LGTM

**Comment by arb** [comment]
Hey @jdtzmn thanks for the PR! Since you're adding this, I assume you have a use case for it right now? What are you trying to validate in cookies? Just some state in there or does this have something to do with authenticating users?

**Comment by jdtzmn** [comment]
One feature that I might add (I don't know if this is a part of Joi or not) is that it is not possible for extra cookies to be declined by the server. For example, if the client requests a route on the server, I'm not sure that the client should be given an error response just because they have extra cookies—the request should only be declined if the necessary cookies are missing or of the wrong type. Not sure how I would implement this though.

**Comment by arb** [comment]
@jdtzmn to allow extra cookies, you'd add [`.unknown()`](https://github.com/hapijs/joi/blob/master/API.md#objectunknownallow) to your cookie schema. I did it one of the tests for headers because there are always extra headers.

I'm a little leary to add cookie support because of the different ways parsed cookies can be represented depending on the parser used. This would require the user of celebreate to know how weird cookies are handled and write schema accordingly.

On the other hand, hapi does this and part of the goal of this lib is to bring hapi-like validation to express. I haven't reviewed the actual code yet as I'm still on the fence if I want to support cookie valiadtion or not. This is the first time this feature has been requested.

**Comment by jdtzmn** [comment]
@arb That sounds reasonable. Just keep me posted as to what you decide.

**Comment by jdtzmn** [comment]
@arb Wait I'm confused. It makes sense that you would not add cookie support, but you have added support for body-parser, which is quite similar in many respects. It is also made and maintained by the same people. Just so I can know for the future, how is cookie-parser support different than having body-parser support?

**Comment by arb** [comment]
Great question @jdtzmn! I supported `req.body` because I think most middleware functions that parse the incoming request payload put it in `req.body`. Since express leaves this kind of thing out in userland code, I can't *KNOW* with certainty where certain values on the request object are going to live. `req.body` seems pretty standard so I rolled the dice on validating there.

Cookies could be under `req.cookie`, `req.cookies`, `req.signedCookies`... I'm not very knowledgable in the express ecosystem to be able to say what cookie parsing middleware is the defacto standard.

This is just speculation on my part. Do people use anything other than the official `cookie-parser` for cookie needs in their express servers?

**Comment by jdtzmn** [comment]
@arb I don’t think that people use anything other than the official cookie-parser. While I am sure that people construct their own, cookie-parser, like body-parser, is maintained by the creators of express and is officially used by most express apps (for example, it is part of the express [middleware documentation](https://expressjs.com/en/resources/middleware/cookie-parser.html)

File: package.json [hunk 1]
```diff
@@ -36,6 +36,8 @@
     "artificial": "0.1.x",
     "benchmark": "2.1.4",
     "body-parser": "1.18.2",
+    "cookie-parser": "^1.4.3",
```
Comment (arb) [inline package.json]: Please pin these to exact versions.

File: test/integration.test.js [hunk 2]
```diff
@@ -12,6 +14,7 @@ const {
 
 const Server = () => {
   const server = Express();
+  server.use(CookieParser('secret'));
```
Comment (arb) [inline test/integration.test.js]: Please use a random string here so we know that the exact string doesn't really matter for the purposes of testing.

**Pull request merged**